app/src/main/java/org/schabi/newpipe/player/playqueue/PlayQueue.java

```diff
@@ -500,22 +519,19 @@ public synchronized boolean previous() {
      * we don't have to do anything with new queue.
      * This method also gives a chance to track history of items in a queue in
      * VideoDetailFragment without duplicating items from two identical queues
-     * */
+     */
     @Override
     public boolean equals(@Nullable final Object obj) {
```

Comment:
Did you autogenerate it or manually implement it, same for hashCode

Reply:
They’re manually implemented. Since `List.equals` calls `equals` for every pair of objects in the list and `PlayQueueItem` overrides `equals` to compare `url`, I simplified it to `streams.equals(other.streams)`.